mods/settlers/settlers.skin.js

```diff
@@ -116,9 +116,35 @@ class SettlersSkin {
 			}
 
 		}
-		return null;
+		return null;	
+	}
+
+	resourceIcon(res){
+		for (let i of this.resources){

```

Comment:
Does this work?:  

```js
const element = this.resources.find(element => element.name === res);
return element?.icon;
```

Or
```js
return this.resources.find(element => element.name === res)?.icon;
```

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find